### ISSUE TITLE
scene_to_screen not defined

### ISSUE THREAD

**Issue by AlexisRenchon** [jkrumbiegel/SankeyMakie.jl#1]
Seems like the code needs some simple fix after https://github.com/MakieOrg/Makie.jl/issues/3013, I tried to do it for a PR but I still got `no method matching plot_to_screen(::Vector{Point{2, Float32}}, ::Scene)`

**Comment by ffreyer** [comment]
The methods for this are here: 
https://github.com/MakieOrg/Makie.jl/blob/ac02141c4c87dbf71d06b301f6dc18f5719e6d05/src/basic_recipes/error_and_rangebars.jl#L246
The main point of these changes was to make these functions more complete, i.e. aware of the space keyword, of transform functions, model transforms. 
I think here you should be able to just replace the old function with `Makie.plot_to_screen(parent_plot, data)` and pass `transformation = Transformation()` in the poly call to avoid applying them twice.

**Comment by AlexisRenchon** [comment]
I'm not good at this... 
```
corners = Makie.plot_to_screen(
                scene,
                Point2f[
                    (x0, y0 - lwidth/2),
                    (x1, y1 - lwidth/2),
                    (x1, y1 + lwidth/2),
                    (x0, y0 + lwidth/2),
                ]
            )
```
I replaced the old function with the code above, I am not sure where to write `transformation = Transformation()` 

https://github.com/AlexisRenchon/SankeyMakie.jl/blob/master/src/SankeyMakie.jl#L410

**Comment by ffreyer** [comment]
You need to pass a plot as the first argument, i.e. the `s` from
https://github.com/jkrumbiegel/SankeyMakie.jl/blob/47f104eb5cb24c2e7267ebde96686da8e7b3b7e2/src/SankeyMakie.jl#L49
You can get the scene from that with `Makie.parent_scene(s)`.
It would also be good to react to `plot.model` and `transform_func(plot)` to be through with updates like here: 
https://github.com/MakieOrg/Makie.jl/blob/ac02141c4c87dbf71d06b301f6dc18f5719e6d05/src/basic_recipes/error_and_rangebars.jl#L200
Without that changing `ax.xscale` or calling `translate!(plot, ...)` wouldn't trigger an update.

> I am not sure where to write transformation = Transformation()

Just in the plot call that uses `corners`. For example:
https://github.com/MakieOrg/Makie.jl/blob/ac02141c4c87dbf71d06b301f6dc18f5719e6d05/src/basic_recipes/bracket.jl#L111

**Comment by simonsteiger** [comment]
I have tried to fix this based on the links @ffreyer posted. The `scene_to_screen not defined` error is gone, but I still seem to be missing an important part of the solution – the resulting polygons seem to not be scaled properly to fit the axis.

![](https://github.com/user-attachments/assets/70622249-f182-4825-9d41-c2d9ad82fbeb)

I would be happy to try to push this over the finishing line and and submit a PR, but would need some more guidance.

This is the commit with my attempt at fixing the issue:
https://github.com/simonsteiger/SankeyMakie.jl/commit/8d6e1b0cd03218efc7c645d6fb605fdbddd35c05

If I had to guess I would say that something with `Transformation()` is going wrong?
